src/python/pants/backend/java/compile/javac_binary.py

```diff
@@ -25,52 +23,16 @@ class JavacBinary:
 
 
 @rule
-async def setup_javac_binary(coursier: Coursier, javac: JavacSubsystem) -> JavacBinary:
-    if javac.options.jdk == "system":
-        process_result = await Get(
-            ProcessResult,
-            Process(
-                argv=[
-                    coursier.coursier.exe,
-                    "java",
-                    "--system-jvm",
-                    "-version",
-                ],
-                input_digest=coursier.digest,
-                description="Invoke Coursier with system-jvm to fingerprint JVM version.",
-                cache_scope=ProcessCacheScope.PER_RESTART_SUCCESSFUL,
```

Comment:
I noticed while rebasing #12982 that this line was lost: it's important when using the system JDK, because it isn't stable, and the fingerprint computation can't be cached. I doubt it's related to the current flakiness, but fyi.